.github/workflows/ci.yml

```diff
@@ -0,0 +1,290 @@
+          cp i18n/en.json /tmp/en.json
+          mkdir -p /tmp/fake-mobile-dir/assets/base/i18n/
+          echo '{}' > /tmp/fake-mobile-dir/assets/base/i18n/en.json
+          npm run mmjstool -- i18n extract-webapp --webapp-dir . --mobile-dir /tmp/fake-mobile-dir
+          diff /tmp/en.json i18n/en.json
+          # Address weblate behavior which does not remove whole translation item when translation string is set to empty
+          npm run mmjstool -- i18n clean-empty --webapp-dir . --mobile-dir /tmp/fake-mobile-dir --check
+          npm run mmjstool -- i18n check-empty-src --webapp-dir . --mobile-dir /tmp/fake-mobile-dir
+          rm -rf tmp
+  check-type:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/setup-node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        id: setup_node
+        with:
+          node-version-file: ".nvmrc"
+          cache: npm
+      - uses: actions/cache@2b8105bdae4b746666ee225970c4a281bbd9d51f # v3.2.4
+        id: npm-cache
+        with:
+          path: |
+            '**/node_modules'
+            'e2e/playwright/node_modules'
+            'e2e/cypress/node_modules'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('e2e/cypress/package-lock.json') }}-${{ hashFiles('e2e/playwright/package-lock.json') }}
+      - name: ci/get-node-modules
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: |
+          make node_modules
+          make e2e/playwright/node_modules
+      - name: ci/lint
+        run: |
+          npm run check-types
+  tests:
+    runs-on: ubuntu-22.04
+    needs: [check-type, check-i18n, check-lint]
+    permissions:
+      checks: write
+      pull-requests: write
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/setup-node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        id: setup_node
+        with:
+          node-version-file: ".nvmrc"
+          cache: npm
+      - uses: actions/cache@2b8105bdae4b746666ee225970c4a281bbd9d51f # v3.2.4
+        id: npm-cache
+        with:
+          path: |
+            '**/node_modules'
+            'e2e/playwright/node_modules'
+            'e2e/cypress/node_modules'
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}-${{ hashFiles('e2e/cypress/package-lock.json') }}-${{ hashFiles('e2e/playwright/package-lock.json') }}
+      - name: ci/get-node-modules
+        if: steps.npm-cache.outputs.cache-hit != 'true'
+        run: |
+          make node_modules
+          make e2e/playwright/node_modules
+      - name: ci/Get number of CPU cores
+        id: cpu-cores
+        uses: SimenB/github-actions-cpu-cores@c508d404ab007faae80a014072fd8c0e17792118 # v1.1.0
```

Comment:
I've reviewed the code for this action. 
https://github.com/SimenB/github-actions-cpu-cores/

It's pretty safe to use, and we are already are SHA pinning it. 

I would prefer to use a public Github Action as blackbox (inputs/outputs), instead of hardcoding some shell commands to detect cores `cat proc/cpuinfo | grep | wc -l` etc.  
Moreover the ghAction It clearly provides context on what it does. 

Let me know @spirosoik . If you are strongly opinionated on this, I can manually detect with Bash.

Reply:
We can always hard code it to 2 cores . Since this is the core count on the Free GIthub Runners

Reply:
Not strongly opinionated, But I'd prefer to keep it dynamically injected during CI Runtime. 

The current execution time in CircleCI is less than 3 mins due to larger runners used (8 cores). We execute this step in 10 minutes due to the default runners GitHub Runners used (2 cores). If there is need to execute this faster, the only change needed, would be to change the `runs_on` directive for this job, and the pipeline will take care of the rest. 

The cost for detecting cores, is negligible (less than 1 sec) 
<img width="1665" alt="Screenshot 2023-02-14 at 11 01 45" src="https://user-images.githubusercontent.com/36594156/218688338-916f7e6a-4727-4e8c-8a89-e9a9f667197c.png">

Reply:
I will echo what Akis mentioned about dynamically inject it. Yeah I checked the code too, seems pretty safe.